sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/windowExpressions.scala

```diff
@@ -633,7 +636,9 @@ case class CumeDist() extends RowNumberLike with SizeBasedWindowFunction {
   group = "window_funcs")
 // scalastyle:on line.size.limit line.contains.tab
 case class NthValue(input: Expression, offset: Expression, ignoreNulls: Boolean)
-    extends AggregateWindowFunction with OffsetWindowFunction with ImplicitCastInputTypes {
+    extends SpecifiedFrameAggregateWindowFunction
```

Comment:
Why we need to extend `SpecifiedFrameAggregateWindowFunction` even though `frame` with `UnspecifiedFrame` is overridden in L649?

Reply:
> Why we need to extend `SpecifiedFrameAggregateWindowFunction` even though `frame` with `UnspecifiedFrame` is overridden in L649?

Replied in https://github.com/apache/spark/pull/31402#discussion_r567526579